File: Demo/Sources/ViewControllers/Dialog/DialogRootViewController.m
//
//  Globus
//

#import "DialogRootViewController.h"
#import "DialogContentViewController.h"

@interface DialogRootViewController ()

@property(nonatomic, weak) IBOutlet GLBListField* listField;
@property(nonatomic) DialogContentViewController* contentViewController;
@property(nonatomic) GLBDialogViewController* dialogViewController;

@end

@implementation DialogRootViewController

#pragma mark - Synthesize

@synthesize contentViewController = _contentViewController;
@synthesize dialogViewController = _dialogViewController;

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _listField.items = @[
        [GLBListFieldItem itemWithTitle:@"Default" value:nil],
        [GLBListFieldItem itemWithTitle:GLBDialogPushAnimationController.glb_className value:GLBDialogPushAnimationController.class],
    ];
}

#pragma mark - Property

- (DialogContentViewController*)contentViewController {
    if(_contentViewController == nil) {
        _contentViewController = [DialogContentViewController instantiate];
    }
    return _contentViewController;
}

- (GLBDialogViewController*)dialogViewController {
    if(_dialogViewController == nil) {
        _dialogViewController = [GLBDialogViewController dialogViewControllerWithContentViewController:self.contentViewController];
        _dialogViewController.backgroundColor = [UIColor.blackColor colorWithAlphaComponent:0.8f];
        _dialogViewController.backgroundBlurred = NO;
        _dialogViewController.contentVerticalAlignment = GLBDialogViewControllerAlignmentVerticalCenter;
        _dialogViewController.contentHorizontalAlignment = GLBDialogViewControllerAlignmentHorizontalCenter;
        _dialogViewController.contentWidthBehaviour = GLBDialogViewControllerSizeBehaviourFill;
        _dialogViewController.contentHeightBehaviour = GLBDialogViewControllerSizeBehaviourFit;
        _dialogViewController.contentInset = UIEdgeInsetsMake(16.0f, 16.0f, 16.0f, 16.0f);
    }
    return _dialogViewController;
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)changedSelectedItem:(id)sender {
    GLBListFieldItem* selected = _listField.selectedItem;
    if(selected != nil) {
        Class animationControllerClass = selected.value;
        if(selected != nil) {
            self.dialogViewController.animationController = [animationControllerClass new];
        } else {
            self.dialogViewController.animationController = nil;
        }
    } else {
        self.dialogViewController.animationController = nil;
    }
}

- (IBAction)pressedShow:(id)sender {
    [self.dialogViewController presentWithCompletion:nil];
}

@end

File: Demo/Sources/ViewControllers/Dialog/DialogRootViewController.swift
//
//  Globus
//

import Globus

class DialogRootViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var listField: GLBListField!
    
    // MARK - Internal property
    
    internal lazy var contentViewController: DialogContentViewController = {
        return DialogContentViewController.instantiate()!
    }()
    internal lazy var dialogViewController: GLBDialogViewController = {
        let vc = GLBDialogViewController.init(contentViewController: self.contentViewController)
        vc.backgroundColor = UIColor.black.withAlphaComponent(0.8)
        vc.isBackgroundBlurred = false
        vc.contentVerticalAlignment = .center
        vc.contentHorizontalAlignment = .center
        vc.contentWidthBehaviour = .fill
        vc.contentHeightBehaviour = .fit
        vc.contentInset = UIEdgeInsets.init(top: 16, left: 16, bottom: 16, right: 16)
        return vc
    }()
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.listField.items = [
            GLBListFieldItem.init(title: "Default", value:nil),
            GLBListFieldItem.init(title: GLBDialogPushAnimationController.glb_className(), value:GLBDialogPushAnimationController.self),
        ]
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func changedSelectedItem(_ sender: Any) {
        if let selected = self.listField.selectedItem {
            if let animationControllerClass = selected.value as? GLBDialogAnimationController.Type {
                self.dialogViewController.animationController = animationControllerClass.init();
            } else {
                self.dialogViewController.animationController = nil
            }
        } else {
            self.dialogViewController.animationController = nil
        }
    }
    
    @IBAction internal func pressedShow(_ sender: Any) {
        self.dialogViewController.present(completion: nil)
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "DialogRootViewController-Swift"
    }
    
}
